### PR TITLE
Issue/8148 > master

### PR DESCRIPTION
**Pull request by davidgrindmill**
- Add faceted search (checkboxes) for Ingredients/Genres to the Product Catalog page.
- Re-enable Product uploader module and corresponding role.

### CODE REVIEW

File: config/sync/views.view.catalog.yml [hunk 1]
```diff
@@ -447,6 +447,10 @@ display:
         type: normal
         title: Catalog
         menu_name: main
+      cache:
+        type: none
+      defaults:
+        cache: false
```
Comment (dzadza) [inline config/sync/views.view.catalog.yml:454]: I don't know I'm just asking, why do you need to turn off caching?

Comment (davidgrindmill) [reply config/sync/views.view.catalog.yml:454]: It wasn't intentional, but I fixed it now (see commit `84c22d8`).

**Review by dzadza — DISMISSED** [review]

File: config/sync/facets.facet.type_of_food.yml [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+  url_processor_handler:
+    processor_id: url_processor_handler
+    weights:
+      pre_query: 50
+      build: 15
+    settings: {  }
+empty_behavior:
+  behavior: text
+  text_format: plain_text
+  text: 'Facet should be here.'
```
Comment (gkurucz) [inline config/sync/facets.facet.type_of_food.yml]: From a UX perspective this message is not too helpful. I think I'd rather hide the facet if there are no results to filter.

Screenshot_20200416_123531

File: config/sync/search_api.index.default_index.yml [hunk 3]
```diff
@@ -2,35 +2,29 @@ uuid: b11493ed-029b-4616-a795-2a64321f5b9c
...
+    - node
+    - taxonomy
 _core:
   default_config_hash: GLmLegZGHtMF6annRqgjizJAoiAip-IT-XOsDMCZfog
 id: default_index
 name: 'Default content index'
 description: 'Default content index created by the Database Search Defaults module'
 read_only: false
 field_settings:
-  field_description:
```
Comment (gkurucz) [inline config/sync/search_api.index.default_index.yml]: I think you might have removed data to be indexed that you shouldn't have. Seems like only the facets work now, the search bar doesn't:

Screenshot_20200416_124219
Screenshot_20200416_124227

**Review by gkurucz — CHANGES_REQUESTED** [review]

File: config/sync/facets.facet.type_of_food.yml (hunk 2 above)
Comment (davidgrindmill) [reply config/sync/facets.facet.type_of_food.yml]: I used this as sort of a debugging tool: to see that the facet is in the right place even if it doesn't work yet, or it's empty. Of course I can hide it if necessary.

File: config/sync/search_api.index.default_index.yml (hunk 3 above)
Comment (davidgrindmill) [reply config/sync/search_api.index.default_index.yml]: Fixed!

**Review by dzadza — APPROVED** [review]
LGTM

File: config/sync/facets.facet.type_of_food.yml (hunk 2 above)
Comment (davidgrindmill) [reply config/sync/facets.facet.type_of_food.yml]: Fixed!

**Review by gkurucz — APPROVED** [review]

**Pull request merged**